src/commands/head.rs

```diff
@@ -1,82 +1,48 @@
-                    .required(true)
-                    .help("Parquet file to read"),
-            )
-            .arg(
-                Arg::with_name("json")
-                    .long("json")
-                    .short("j")
-                    .takes_value(false)
-                    .required(false)
-                    .help("Use JSON lines format for printing"),
-            )
-            .arg(
-                Arg::with_name("records")
-                    .long("records")
-                    .short("n")
-                    .default_value("5")
-                    .takes_value(true)
-                    .required(false)
-                    .help("The number of records to show (default: 5)"),
-            )
-    }
+/// Prints the first n records of the Parquet file
+#[derive(Parser, Debug)]
+pub struct HeadCommandArgs {
+    /// Use CSV format for printing
+    #[clap(short, long, conflicts_with = "json")]
+    csv: bool,
 
-    pub(crate) fn new(matches: &'a ArgMatches<'a>) -> Self {
-        Self {
-            file_name: matches.value_of("file").unwrap(),
-            num_records: matches.value_of("records").unwrap().parse().unwrap(),
-            format: if matches.is_present("json") {
-                &Formats::Json
-            } else {
-                &Formats::Default
-            },
-        }
-    }
-}
+    /// Use JSON lines format for printing
+    #[clap(short, long, conflicts_with = "csv")]
+    json: bool,
 
-impl<'a> PQRSCommand for HeadCommand<'a> {
-    fn execute(&self) -> Result<(), PQRSError> {
-        // print debugging information
-        debug!("{:#?}", self);
+    /// The number of records to show (default: 5)
+    #[clap(short = 'n', long, default_value = "5")]
+    records: usize,
 
-        if !check_path_present(self.file_name) {
-            return Err(FileNotFound(String::from(self.file_name)));
-        }
-
-        let file = open_file(self.file_name)?;
-        print_rows(file, Some(self.num_records), self.format)?;
+    /// Parquet file to read
+    file: PathBuf,
+}
 
-        Ok(())
+pub(crate) fn execute(opts: HeadCommandArgs) -> Result<(), PQRSError> {
+    let format = if opts.json {
+        Formats::Json
+    } else if opts.csv {
+        Formats::Csv
+    } else {
+        Formats::Default
+    };
+
+    debug!("The file name to read is: {}", opts.file.display());
+    debug!("Number of records to print: {}", opts.records);
+    debug!("Use Output format: {}", format);
+
+    if !check_path_present(&opts.file) {
+        return Err(FileNotFound(opts.file));
     }
-}
 
-impl<'a> fmt::Debug for HeadCommand<'a> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        writeln!(f, "The file name to read is: {}", &self.file_name)?;
-        writeln!(f, "Number of records to print: {}", &self.num_records)?;
-        writeln!(f, "Use Output format: {}", self.format.to_string())?;
+    let file = open_file(&opts.file)?;
+    print_rows(file, Some(opts.records), format)?;
```

Comment:
I was testing with a JSON file and the update seems to introduce a new regression with some files:

```
Error: ArrowReadWriteError(JsonError("data type Decimal(12, 6) not supported in nested map for json writer"))
```

Doesn't have an issue with the old implementation